src/content/en/fundamentals/instant-and-offline/pwa-architecture/index.md

```diff
@@ -0,0 +1,922 @@
+  </tr>
+  <tr>
+    <td>dynamic content
+</td>
+    <td>Dynamic content is all of the data, images, and other resources that your web app needs to function, but exists independently from your app shell. Although the app shell is intended to quickly populate the content of your site, users might expect dynamic content, in which case your app must fetch data specific to the user’s needs. Sometimes an app pulls this data from external, third-party APIs, and sometimes from first-party data that is dynamically generated or frequently updated.</td>
+  </tr>
+  <tr>
+    <td>Fetch API</td>
+    <td>You can optionally use the <a href="https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API">Fetch API</a> to fetch resources. For example, if your web app is for a newspaper, it might make use of a first-party API to fetch recent articles, and a third-party API to fetch the current weather. Both of those types of requests fall into the category of dynamic content.</td>
+  </tr>
+  <tr>
+    <td>progressive enhancement</td>
+    <td>An approach to web development that begins with common browser features, and then adds in functionality or enhancements when the user’s browser supports more modern technologies.</td>
+  </tr>
+  <tr>
+    <td>PWA architecture styles</td>
+    <td>Any of several approaches to building PWAs based on the back-end technologies available and the performance requirements. The patterns include using an app shell, server-side rendering, client-side rendering, and others. These patterns are listed in Choose an Architecture.  </td>
+  </tr>
+  <tr>
+    <td>server-side rendering (SSR)</td>
+    <td>SSR means when the browser navigates to a URL fetches the page, it immediately gets back HTML describing the page. SSR is nice because the page loads faster (this can be a server-rendered version of the full page, just the app shell or the content). There's no white page displayed while the browser downloads the rendering code and data and runs the code.  If rendering content on the server-side, users can get meaningful text on their screens even if a spotty network connection prevents assets like JavaScript from being fully fetched and parsed. SSR also maintains the idea that pages are documents, and if you ask a server for a document by its URL, the text of the document is returned, rather than a program that generates that text using a complicated API.</td>
+  </tr>
+  <tr>
+    <td>service worker</td>
+    <td>A type of web worker that runs alongside your web app but with a life span that is tied to the execution of the app’s events. Some of its services include a network proxy written in JavaScript that intercepts HTTP/HTTPS requests made from web pages. It also receives push messages. Additional features are planned in the future.</td>
+  </tr>
+  <tr>
+    <td>sw-precache</td>
```

Comment:
You should probably clarify that these are tools to simplify implementation and not requirements.